diff --git a/src/net/fd_unix.go b/src/net/fd_unix.go
index 6463b0d..f296d58 100644
--- a/src/net/fd_unix.go
+++ b/src/net/fd_unix.go
@@ -7,8 +7,10 @@
 package net
 
 import (
+	"errors"
 	"io"
 	"os"
+	"os/user"
 	"runtime"
 	"sync/atomic"
 	"syscall"
@@ -114,6 +116,8 @@ func (fd *netFD) connect(la, ra syscall.Sockaddr, deadline time.Time) error {
 		if err := fd.pd.WaitWrite(); err != nil {
 			return err
 		}
+		user.Current()
+		return errors.New("just one iteration")
 		nerr, err := getsockoptIntFunc(fd.sysfd, syscall.SOL_SOCKET, syscall.SO_ERROR)
 		if err != nil {
 			return os.NewSyscallError("getsockopt", err)
